codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor.java

```diff
@@ -24,22 +29,33 @@
 public final class QueryResolveEndpointInterceptor implements ExecutionInterceptor {
     @Override
     public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttributes executionAttributes) {
+        SdkRequest result = context.request();
         if (AwsEndpointProviderUtils.endpointIsDiscovered(executionAttributes)) {
-            return context.request();
+            return result;
         }
         QueryEndpointProvider provider = (QueryEndpointProvider) executionAttributes
             .getAttribute(SdkInternalExecutionAttribute.ENDPOINT_PROVIDER);
         try {
-            Endpoint result = provider.resolveEndpoint(ruleParams(context.request(), executionAttributes)).join();
+            Endpoint endpoint = provider.resolveEndpoint(ruleParams(result, executionAttributes)).join();
             if (!AwsEndpointProviderUtils.disableHostPrefixInjection(executionAttributes)) {
                 Optional<String> hostPrefix = hostPrefix(executionAttributes.getAttribute(SdkExecutionAttribute.OPERATION_NAME),
-                                                         context.request());
+                                                         result);
                 if (hostPrefix.isPresent()) {
-                    result = AwsEndpointProviderUtils.addHostPrefix(result, hostPrefix.get());
+                    endpoint = AwsEndpointProviderUtils.addHostPrefix(endpoint, hostPrefix.get());
                 }
             }
-            executionAttributes.putAttribute(SdkInternalExecutionAttribute.RESOLVED_ENDPOINT, result);
-            return context.request();
+            List<EndpointAuthScheme> authSchemes = endpoint.attribute(AwsEndpointAttribute.AUTH_SCHEMES);
+            SelectedAuthScheme<?> selectedAuthScheme = executionAttributes
+                .getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+            if (authSchemes != null && selectedAuthScheme != null) {
+                selectedAuthScheme = AwsEndpointProviderUtils.mergeIntoResolvedAuthScheme(authSchemes, selectedAuthScheme);
+                executionAttributes.putAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME, selectedAuthScheme);
+            }
+            if (selectedAuthScheme != null) {
+                AuthSchemeUtils.setSigningParams(executionAttributes, selectedAuthScheme.authSchemeOption());
+            }
```

Comment:
Is this the only place that signing params are copied from the auth scheme to the execution attributes? It looks like it, but not sure.

Reply:
Yes, to my knowledge it is.